src/Akka.Persistence.Linq2Db.Data.Compatibility.Tests/DataCompatibilitySpec.cs

```diff
@@ -0,0 +1,91 @@
+﻿// -----------------------------------------------------------------------
+//  <copyright file="DataCompatibilitySpec.cs" company="Akka.NET Project">
+//      Copyright (C) 2013-2022 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Akka.Hosting;
+using Akka.Persistence.Linq2Db.Data.Compatibility.Tests.Internal;
+using Akka.Persistence.Query;
+using Akka.Persistence.Sql.Compat.Common;
+using Akka.Persistence.Sql.Linq2Db.Query;
+using Akka.Streams;
+using Akka.Streams.Dsl;
+using Xunit;
+using Xunit.Abstractions;
+using FluentAssertions;
+
+namespace Akka.Persistence.Linq2Db.Data.Compatibility.Tests
+{
+    // Default compatibility spec, logical delete turned off
+    public abstract class DataCompatibilitySpec<T> : DataCompatibilitySpecBase<T> where T: ITestContainer, new()
+    {
+        protected DataCompatibilitySpec(ITestOutputHelper output): base(output)
+        {
+        }
+
+        protected sealed override void Setup(AkkaConfigurationBuilder builder, IServiceProvider provider)
+        {
+        }
+        
+        [Fact(DisplayName = "Linq2Db should recover data created using other persistence plugins")]
+        public async Task RecoveryValidationTest()
+        {
+            await ValidateRecovery(TestCluster!.ShardRegions[0]);
+        }
+
+        [Fact(DisplayName = "Linq2Db should be able to retrieve persistence ids on data created by other persistence plugin")]
+        public async Task PersistenceIdQueryValidationTest()
```

Comment:
LGTM - but should add a comment explaining how many IDs are stored in the database (be explicit)

Reply:
On a related note, we should make sure that we are handling 'loop-overs' on these (or other persistence.query) tests, i.e. make sure we are doing enough events to where multiple batches to store are required.

Reply:
Could you help me understand this better? Running repeated queries over the same data set or running multiple queries concurrently?

Reply:
I could have worded that better.

Uhh, long story short, when we do reads on the DB, we chunk them. I'd have to look to find which number it is (I haven't ever really used the query bits) but if I had to guess it's the `replay-batch-size`. 

tl;dr- if we are aiming to handle compatibility as well as general coverage I would encourage making sure there's enough events that the query stream does at least two hits to the DB to fetch them all.

Reply:
Can we limit the batch fetch to less than the data in the database? The data is a known variable, each entity would have 12 events in the journal since the last snapshot.
So if we limit the batch fetch to 6, that would force Linq2Db to do 2 chunks of fetches?

Reply:
Yep, making the batch size artificially smaller for testing is a good solution here, to ensure that pagination works. Let's give that a try.

Reply:
@Arkatufus I will be honest and say I don't know which setting is correct off the top of my head, but you could probably turn all of them down for the sake of the compatibility tests.